chain/processor.go

```diff
@@ -100,6 +110,7 @@ func (b *StatelessBlock) Execute(
 					toCache[k] = &fetchData{v, true, numChunks}
 					storage[k] = v
 				}
+				cacheLock.RUnlock()
```

Comment:
Even if we did want to hold the lock during the entirety of this disk fetch, we wouldn't prevent multiple fetches of the same key because we haven't updated `cache` by the time this lock is released.